MonoGame.Framework/Random.cs

```diff
@@ -0,0 +1,254 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.Xna.Framework
+{
+    public static partial class MathHelper
+    {
+        /// <summary>
+        /// Class for generating random numbers consistantly when using Mono or Microsoft compilers
+        /// </summary>
+        public class Random
+        {
+            #region Static methods
+
+            [ClsCompliant(false)]
+            public static ulong GenerateSeed()
+            {
+                byte[] guid = new Guid().ToByteArray();
+                return (BitConverter.ToUInt64(guid, 0) ^ BitConverter.ToUInt64(guid, 7));
+            }
+
+            #endregion
+
+            #region Private Fields
+
+            private ulong _seed;
+            private ulong _lastValue;
+
+            #endregion
+
+            #region Properties
+
+            [ClsCompliant(false)]
+            public ulong Seed
+            {
+                get
+                {
+                    return this._seed;
+                }
+                set
+                {
+                    this._seed = value;
+                    this._lastValue = value;
```

Comment:
> But you would want to be reading the seed for the next get of a random number

Right.  The cases I specified before all want the current state of the randomizer so it can restart from there if need be.  (oh... replays is another case where you might store the random state) 

Having the starting seed could maybe be useful in some cases, but it is easy for the developer to just store that himself.

So I agree with you.